main/modes/mode_jukebox.c

```diff
@@ -0,0 +1,750 @@
+                        }
+                        jukebox->categoryIdx = jukebox->categoryIdx - 1;
+
+                        jukebox->songIdx = 0;
+                        break;
+                    }
+                    case DOWN:
+                    {
+                        uint8_t length;
+                        if(jukebox->inMusicSubmode)
+                        {
+                            length = lengthof(musicCategories);
+                        }
+                        else
+                        {
+                            length = lengthof(sfxCategories);
+                        }
+
+                        jukebox->categoryIdx = (jukebox->categoryIdx + 1) % length;
+
+                        jukebox->songIdx = 0;
+                        break;
+                    }
+                    case LEFT:
+                    {
+                        uint8_t length;
+                        if(jukebox->inMusicSubmode)
+                        {
+                            length = musicCategories[jukebox->categoryIdx].numSongs;
+                        }
+                        else
+                        {
+                            length = sfxCategories[jukebox->categoryIdx].numSongs;
+                        }
+
+                        if(jukebox->songIdx == 0)
+                        {
+                            jukebox->songIdx = length;
+                        }
+                        jukebox->songIdx = jukebox->songIdx - 1;
+                        break;
+                    }
+                    case RIGHT:
+                    {
+                        uint8_t length;
+                        if(jukebox->inMusicSubmode)
+                        {
+                            length = musicCategories[jukebox->categoryIdx].numSongs;
+                        }
+                        else
+                        {
+                            length = sfxCategories[jukebox->categoryIdx].numSongs;
+                        }
+
+                        jukebox->songIdx = (jukebox->songIdx + 1) % length;
+                        break;
+                    }
+                    default:
+                    {
+                        break;
+                    }
+                }
+            }
+            break;
+        }
+    }
+}
+
+/**
+ * Update the display by drawing the current state of affairs
+ */
+void  jukeboxMainLoop(int64_t elapsedUs)
+{
+    jukebox->disp->clearPx();
+    switch(jukebox->screen)
+    {
+        case JUKEBOX_MENU:
+        {
+            drawMeleeMenu(jukebox->disp, jukebox->menu);
+            break;
+        }
+        case JUKEBOX_PLAYER:
+        {
+            jukeboxLedDances[jukebox->danceIdx].func(elapsedUs, jukeboxLedDances[jukebox->danceIdx].arg, jukebox->resetDance);
+            jukebox->resetDance = false;
+
+            fillDisplayArea(jukebox->disp, 0, 0, jukebox->disp->w, jukebox->disp->h, c010);
```

Comment:
Filling the whole area after calling `clearPx()` is a bit redundant. You can move `clearPx()` to just the `JUKEBOX_MENU` case.

Reply:
done